### PR TITLE
prevent undefined behaviour of SPIR-V Backend non-asserts builds when dealing with token type

### PR DESCRIPTION
**Pull request by VyacheslavLevytskyy**
The goal of this PR is to fix the issue when use of token type in LLVM intrinsic causes undefined behavior of SPIR-V Backend code generator when assertions are disabled: https://github.com/llvm/llvm-project/issues/78434

Among possible fix options, discussed in the https://github.com/llvm/llvm-project/issues/78434 issue description, the option to generate a meaningful error before execution arrives at the `llvm_unreachable` call looks like a better solution for now, because SPIR-V doesn't support token type anyway without additional extensions.

The PR is to generate a user-friendly error message and exit without generating a stack dump when such a usage of token type was detected that would lead to undefined behavior of SPIR-V Backend code generator.

### CODE REVIEW

**Comment by VyacheslavLevytskyy** [comment]
@michalpaszkowski Can you please have a look at this one as well when you have time. Thank you!

**Review by Keenuts — APPROVED** [review]
LGTM on my end. Thanks!

File: llvm/test/CodeGen/SPIRV/token/token_type_preallocated_setup_arg.ll [hunk 1]
```diff
@@ -0,0 +1,17 @@
+; Example of token usage is from https://llvm.org/docs/LangRef.html (Preallocated Operand Bundles)
```
Comment (michalpaszkowski) [inline llvm/test/CodeGen/SPIRV/token/token_type_preallocated_setup_arg.ll:1]: Please change the name to use dashes instead of underscores

**Review by michalpaszkowski — APPROVED** [review]
LGTM!

**Pull request merged**